File: Java/Dagger/atm/src/main/java/ro/razvan/java/dagger/atm/AmountsModule.java
package ro.razvan.java.dagger.atm;

import dagger.Module;
import dagger.Provides;

import java.math.BigDecimal;

@Module
public interface AmountsModule {

    @Provides
    @MaximumWithdrawal
    static BigDecimal provideMaximumWithdrawal() {
        return new BigDecimal(1_000);
    }

    @Provides
    @MinimumBalance
    static BigDecimal provideMinimumBalance() {
        return BigDecimal.ZERO;
    }
}

File: Kotlin/Dagger/atm/src/main/kotlin/ro/razvan/kotlin/dagger/atm/AmountsModule.kt
package ro.razvan.kotlin.dagger.atm

import dagger.Module
import dagger.Provides
import java.math.BigDecimal

@Module
object AmountsModule {

    @Provides
    @MaximumWithdrawal
    fun provideMaximumWithdrawal(): BigDecimal = BigDecimal(1_000)

    @Provides
    @MinimumBalance
    fun provideMinimumBalance(): BigDecimal = BigDecimal.ZERO

}